x-pack/solutions/security/plugins/security_solution/public/plugin_contract.ts

```diff
@@ -14,24 +13,31 @@ import { navLinks$, updateNavLinks } from './common/links/nav_links';
 import { breadcrumbsNav$ } from './common/breadcrumbs';
 import { ContractComponentsService } from './contract_components';
 import { OnboardingService } from './onboarding/service';
+import { ProductFeatureKeyType } from '../../../packages/features/src/types';
 
 export class PluginContract {
   public componentsService: ContractComponentsService;
   public upsellingService: UpsellingService;
   public onboardingService: OnboardingService;
   public isSolutionNavigationEnabled$: BehaviorSubject<boolean>;
+  public productFeatureKeys$: BehaviorSubject<Set<ProductFeatureKeyType> | null>;
 
   constructor(private readonly experimentalFeatures: ExperimentalFeatures) {
     this.onboardingService = new OnboardingService();
     this.componentsService = new ContractComponentsService();
     this.upsellingService = new UpsellingService();
     this.isSolutionNavigationEnabled$ = new BehaviorSubject<boolean>(false); // defaults to classic navigation
+    this.productFeatureKeys$ = new BehaviorSubject<Set<ProductFeatureKeyType> | null>(null);
   }
 
   public getSetupContract(): PluginSetup {
+
     return {
       resolver: lazyResolver,
       experimentalFeatures: { ...this.experimentalFeatures },
+      setProductFeatureKeys: (productFeatureKeys: Set<ProductFeatureKeyType>) => {
+        this.productFeatureKeys$.next(productFeatureKeys);
+      },
```

Comment:
Consider centralize the Set creation here:
```suggestion
      setProductFeatureKeys: (productFeatureKeys: ProductFeatureKeys) => {
        this.productFeatureKeys$.next(new Set(productFeatureKeys));
      },
```